projects/longevity/inference/inference/main.py

```diff
@@ -0,0 +1,141 @@
+import logging
+from pathlib import Path
+from typing import List
+
+from infer.deploy import main as deploy_infer
+from typeo import scriptify
+
+from aframe.logging import configure_logging
+
+
+@scriptify
+def main(
+    basedir: Path,
+    original_model_repo_dir: str,
+    image: str,
+    model_name: str,
+    accounting_group: str,
+    accounting_group_user: str,
+    Tb: float,
+    shifts: List[float],
+    sample_rate: float,
+    inference_sampling_rate: float,
+    ifos: List[str],
+    batch_size: int,
+    integration_window_length: float,
+    cluster_window_length: float,
+    psd_length: float,
+    fduration: float,
+    throughput: float,
+    chunk_size: float,
+    sequence_id: int,
+    model_version: int = -1,
+    verbose: bool = False,
+):
+    configure_logging(basedir / "infer.log", verbose=verbose)
+    # loop over intervals
+    intervals = [
+        x
+        for x in basedir.iterdir()
+        if x.is_dir() and not x.name.startswith(".")
+    ]
+
+    for interval in intervals:
+        datadir = interval / "data"
+        injection_set_file = datadir / "test" / "waveforms.h5"
+
+        # launch inference job for each interval
+        # analyzing data with the retrained models
+        retrained = interval / "retrained"
+        infer_dir = retrained / "infer"
+        infer_dir.mkdir(exist_ok=True, parents=True)
+        complete = all(
```

Comment:
Is this just so that you don't accidentally overwrite these files?

Reply:
just a caching check so we don't re run if these already exist